db_stress_tool/db_stress_test_base.cc

```diff
@@ -3171,6 +3170,12 @@ void CheckAndSetOptionsForUserTimestamp(Options& options) {
     fprintf(stderr, "Bulk loading may not support timestamp yet.\n");
     exit(1);
   }
+  if (from_restore) {
```

Comment:
It seems we need all and only these checks for restore, maybe it's better to move these asserts to PrepareOptionsForRestoredDB and remove `from_restore`.

Reply:
This block has an assertion that compare `options.comparator` and `cmp` are the same static object coming from object registry. Moving the check back to `PrepareOptionsForRestoredDB` cannot have this check. Do you mean the assertion should just be removed?

Reply:
I meant to move asserts to `PrepareOptionsForRestoredDB()`. 

> cannot have this check

Can we do `assert(options.comparator == test::BytewiseComparatorWithU64TsWrapper())` in `PrepareOptionsForRestoredDB()`?

Reply:
😄 good point. Assertions moved.